Docker-compose branch isolation may not work for valid branch names

<!--

Note that anything written between these symbols will not appear in the actual, published issue. They serve as instructions for filling out this template.  Please use the 'preview' tab above this textbox to verify formatting before submitting.

Instructions:
- Start by replacing the content in "[Title]" and give a "[Brief description]" of the issue above
- Please remove any unused, optional sections below.

-->

## Observed behavior
<!-- Briefly describe the current behavior, including screenshots or other references when applicable -->
If the branch name has a slash, [the branch isolation](https://github.com/learningequality/studio/blob/601ddc0928af280f088cf165dad2b402f03d674b/Makefile#L139) for docker volumes doesn't work:
```
# build all studio docker image and all dependent services using docker-compose
docker compose build
invalid project name "studio_pr/Jovian-Dsouza/4384": must consist only of lowercase alphanumeric characters, hyphens, and underscores as well as start with a letter or number
make: *** [Makefile:169: dcbuild] Error 15
```

## Expected behavior
<!-- Briefly describe the expected behavior that did not occur -->
Characters that are valid for branch names are replaced (perhaps with `-`) such that the above error does not occur.

## User-facing consequences
<!-- Implications and real-world consequences for learners, coaches, admins, and other users of the application -->
Contributor confusion on why this is happening when following our development guidance.

--- a/Makefile
+++ b/Makefile
@@ -136,7 +136,9 @@ dummyusers:
 hascaptions:
 	python contentcuration/manage.py set_orm_based_has_captions
 
-export COMPOSE_PROJECT_NAME=studio_$(shell git rev-parse --abbrev-ref HEAD)
+BRANCH_NAME := $(shell git rev-parse --abbrev-ref HEAD | sed 's/[^a-zA-Z0-9_-]/-/g')
+
+export COMPOSE_PROJECT_NAME=studio_$(BRANCH_NAME)
 
 purge-postgres: .docker/pgpass
 	-PGPASSFILE=.docker/pgpass dropdb -U learningequality "kolibri-studio" --port 5432 -h localhost

